src/App.tsx

```diff
@@ -1,12 +1,48 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
+import * as goodsAPI from './api/goods';
+import GoodsList from './components/GoodsList';
+import Button from './components/Button';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+const App: React.FC = () => {
+  const [goods, setGoods] = useState <Good[]>([]);
 
-const App: React.FC = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+  const getAllGoods = async () => {
+    const goodsFromServer = await goodsAPI.getAll();
+
+    setGoods(goodsFromServer);
+  };
+
+  const get5FirstGoods = async () => {
+    const goodsFromServer = await goodsAPI.get5First();
+
+    setGoods(goodsFromServer);
+  };
+
+  const getRedGoods = async () => {
+    const goodsFromServer = await goodsAPI.getRedGoods();
+
+    setGoods(goodsFromServer);
+  };
```

Comment:
these functions look almost the same, let's create one function that receives callback, call it and write the result to the state